bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -939,7 +953,7 @@ public void run() {
                             shouldFlush = true;
                             flushMaxWaitCounter.inc();
                         } else if (qe != null
-                                && ((bufferedEntriesThreshold > 0 && toFlush.size() > bufferedEntriesThreshold)
```

Comment:
you should use `toFlush.size()` here. otherwise you might potentially have a list accumulating a lot of nulls

Reply:
make sure you have a test case cover here

Reply:
changed, will add tests ASAP